packages/mask/src/plugins/RedPacket/SNSAdaptor/SelectNftTokenDialog.tsx

```diff
@@ -192,20 +207,99 @@ const useStyles = makeStyles()((theme) => ({
         width: 64,
         height: 64,
     },
+    selectedTokenAmount: {
+        color: props.isSelectSharesExceed ? '#FF5F5F' : '#1C68F3',
+    },
+    totalAmount: {
+        paddingLeft: 1,
+    },
+    selectAmountBox: {
+        display: 'flex',
+        flexDirection: 'row-reverse',
+        alignItems: 'center',
+    },
+    questionMarkIcon: {
+        padding: 2,
+        width: 12,
+        border: `1px solid ${theme.palette.mode === 'light' ? '#0F1419' : '#D9D9D9'}`,
+        borderRadius: 999,
+        transform: 'translateY(-1px)',
+        height: 12,
+        marginLeft: 5,
+        cursor: 'pointer',
+    },
+    selectBar: {
+        display: 'flex',
+        alignItems: 'center',
+        justifyContent: 'space-between',
+        marginTop: 12,
+        padding: '0 8px',
+    },
+    selectAll: {
+        display: 'flex',
+        alignItems: 'center',
+        fontSize: 14,
+    },
+    selectAllCheckBox: {
+        display: 'flex',
+        justifyContent: 'center',
+        alignItems: 'center',
+        cursor: 'pointer',
+        width: 17,
+        height: 17,
+        borderRadius: 6,
+        marginRight: 5,
+        border: '2px solid #6E748E',
+        backgroundColor: 'white',
+    },
+    selectAllCheckBoxText: {
+        pointerEvents: 'none',
+    },
+    selectSharesExceed: {
+        color: '#FF5F5F',
+    },
+    selectSharesExceedBox: {
+        display: 'flex',
+        alignItems: 'center',
+        justifyContent: 'space-between',
+        margin: '14px 4px',
+    },
+    arrow: {
+        color: theme.palette.mode === 'dark' ? '#fff' : '#111418',
+        transform: 'translateX(260px) !important',
+    },
+    tooltip: {
+        transform: 'translateX(20px) !important',
+        padding: '10px 20px',
+        width: 256,
+        backgroundColor: theme.palette.mode === 'dark' ? '#fff' : '#111418',
+    },
+    tooltipText: {
+        color: theme.palette.mode === 'dark' ? '#000' : '#fff',
+    },
+    tokenId: {
+        color: '#1D9BF0',
+    },
+    nonExistedTokenBox: {
+        height: 388,
+        width: '100%',
+        display: 'flex',
+        justifyContent: 'center',
+        alignItems: 'center',
+    },
 }))
 
 export interface SelectNftTokenDialogProps extends withClasses<never> {
     open: boolean
     loadingOwnerList: boolean
     onClose: () => void
     contract: ERC721ContractDetailed | undefined
-    existTokenDetailedList: ERC721TokenDetailed[]
-    tokenDetailedOwnerList: ERC721TokenDetailed[]
-    setExistTokenDetailedList: React.Dispatch<React.SetStateAction<ERC721TokenDetailed[]>>
+    existTokenDetailedList: (ERC721TokenDetailed & { index: number })[]
```

Comment:
I have a good name to suggest
```ts
type OrderedERC721Token = ERC721TokenDetailed & { index: number}
```

But I don't think it's good to mix index into the token.